packages/sources/dxfeed-test/src/endpoint/price-ws.ts

```diff
@@ -0,0 +1,157 @@
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: DXFeedMessage
+  }
+}
+
+const META_HANDSHAKE = '/meta/handshake'
+const META_CONNECT = '/meta/connect'
+const SERVICE_SUB = '/service/sub'
+const SERVICE_DATA = '/service/data'
+
+const tickerIndex = 0
+const priceIndex = 6
+
+class DxFeedWebsocketTransport extends WebSocketTransport<EndpointTypes> {
+  private _connectionClientId = ''
+  id = 1
+
+  get connectionClientId() {
+    return this._connectionClientId
+  }
+
+  set connectionClientId(id) {
+    this._connectionClientId = id
+  }
+  get handshakeMsg() {
+    return [
+      {
+        id: this.id,
+        version: '1.0',
+        minimumVersion: '1.0',
+        channel: META_HANDSHAKE,
+        supportedConnectionTypes: ['websocket', 'long-polling', 'callback-polling'],
+        advice: {
+          timeout: 60000,
+          interval: 0,
+        },
+      },
+    ]
+  }
+  get firstHeartbeatMsg() {
+    return [
+      {
+        id: ++this.id,
+        clientId: this.connectionClientId,
+        channel: META_CONNECT,
+        connectionType: 'websocket',
+        advice: {
+          timeout: 60000,
+        },
+      },
+    ]
+  }
+  get heartbeatMsg() {
+    return [
+      {
+        id: ++this.id,
+        clientId: this.connectionClientId,
+        channel: META_CONNECT,
+        connectionType: 'websocket',
+      },
+    ]
+  }
+}
+
+export const wsTransport: DxFeedWebsocketTransport = new DxFeedWebsocketTransport({
+  url: (context) => context.adapterConfig.WS_API_ENDPOINT || '',
+  handlers: {
+    open(connection) {
+      return new Promise((resolve) => {
+        connection.on('message', (data: WebSocket.MessageEvent) => {
+          const message: DXFeedMessage[0] = JSON.parse(data.toString())[0]
+          if (message.clientId && message.channel === '/meta/handshake') {
+            wsTransport.connectionClientId = message.clientId
+            connection.send(JSON.stringify(wsTransport.firstHeartbeatMsg))
+          }
+
+          if (message.channel === '/meta/connect') {
+            connection.send(JSON.stringify(wsTransport.heartbeatMsg))
+            resolve()
+          }
+        })
+        connection.send(JSON.stringify(wsTransport.handshakeMsg))
+      })
+    },
+    message(message) {
+      //If dxfeed errors there is no information about failed feeds/params in the message, returning empty array
+      if (message[0].successful === false) {
+        logger.warn('Dxfeed returned unsuccessful message')
+        return []
+      }
+
+      if (Array.isArray(message) && message[0].channel === SERVICE_DATA) {
+        const base = message[0].data[1][tickerIndex]
+        const price = message[0].data[1][priceIndex]
+        return [
+          {
+            params: { base },
+            response: {
+              result: price,
+              data: {
+                result: price,
+              },
+            },
+          },
+        ]
+      } else {
+        return []
+      }
```

Comment:
Can we flip this and make the left path the happy path?

```suggestion
      if (!Array.isArray(message) || message[0].channel !== SERVICE_DATA) {
        return []
      ...
```